### PR TITLE
[IN-1686] Install community requested tools

### PR DESCRIPTION
**Pull request by fehersanyi-bitrise**
### New Pull Request Checklist

*Please mark the points which you did / accept.*

- [ ] The tool I added is stable, and __does not require frequent updates__. _Preinstalled tools on the LTS stacks
  are not updated, so if the tool requires frequent updates you should handle the installation on-demand (see: [Install Any Additional Tool - DevCenter](https://bitrise-io.github.io/devcenter/tips-and-tricks/install-additional-tools/) for more information)_
- [ ] I added a version report line to [`system_report.sh`](/system_report.sh) for the new tool(s) I added
- [ ] I have added an entry to the CHANGELOG.md with a revisionID (the current date and the number of the change e.g.: 2019_10_11_1) and a brief description of the actual change
- [ ] I have updated the revisionID in [bitrise_profile](roles/profiles/files/bitrise_profile) **BITRISE_OSX_STACK_REV_ID** with the one I just created in the CHANGELOG.md

### CODE REVIEW

File: roles/weekly-update-shared/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+gem_packages:
+  - name: "xcpretty-json-formatter"
+    version: "0.1.1"
+  - name: "xcbeautify"
+    version: "0.8.1"
+  - name: "xcbeautify"
+    version: "0.8.1"
```
Comment (MrBlaise) [inline roles/weekly-update-shared/defaults/main.yml:8]:
```suggestion
  - name: "xcbeautify"
    version: "0.8.1"
```

Duplicate packages

File: roles/weekly-update-shared/tests/test_weekly_update_shared.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+def test_maxfiles_exists(host):
+    maxfiles = host.file("/Library/LaunchDaemons/imit.maxfiles.plist")
```
Comment (MrBlaise) [inline roles/weekly-update-shared/tests/test_weekly_update_shared.py:2]:
```suggestion
    maxfiles = host.file("/Library/LaunchDaemons/limit.maxfiles.plist")
```

File: roles/weekly-update-shared/tests/test_weekly_update_shared.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    assert maxfiles.contains("root")
+    assert maxfiles.user == "root"
+    assert maxfiles.group == "wheel"
+    assert maxfiles.mode == 644
+
+def test_maxproc_exists(host):
+    maxproc = host.file("/Library/LaunchDaemons/imit.maxproc.plist")
```
Comment (MrBlaise) [inline roles/weekly-update-shared/tests/test_weekly_update_shared.py:9]:
```suggestion
    maxproc = host.file("/Library/LaunchDaemons/limit.maxproc.plist")
```

**Review by MrBlaise — CHANGES_REQUESTED** [review]
see my comments

**Comment by peterheja-bitrise** [comment]
Please, do not merge it until the end of weekly updates

**Comment by fehersanyi-bitrise** [comment]
@MrBlaise @peter-bitrise my dudes this is still work in progress

**Comment by MrBlaise** [comment]
> @MrBlaise @peter-bitrise my dudes this is still work in progress

@fehersanyi-bitrise You can convert the PR to draft (under the reviewers section)

**Comment by fehersanyi-bitrise** [comment]
Sorry for the big PR :/

File: roles/bitrise-cli/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  file:
+    path: /usr/local/bin/stepman
+    state: absent
+
+- name: clean up previous 'envman'
+  file:
+    path: /usr/local/bin/envman
+    state: absent
+
+- name: Lookup ansible_distribution
```
Comment (bitrise-tom) [inline roles/bitrise-cli/tasks/main.yml:21]: here you can use `ansible_system` that is builtin to ansible's variables

File: roles/bitrise-cli/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
+---
+# ----------------------------------------
+# --- Bitrise CLI tools ---
+# ----------------------------------------
+# bitrise CLI: cleanup previous tool versions
+- name: clean up previous 'bitrise'
```
Comment (bitrise-tom) [inline roles/bitrise-cli/tasks/main.yml:6]: I don't like these three removal steps. 
I understand the logic behind them, but I think they are unnecessary.

The following downloads should work, and if they don't we should have an molecule test that runs the cli and makes sure the correct version is installed

File: roles/bitrise-cli/tests/test_bitrise_cli.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
+def test_bitrise_cli_exists(host):
```
Comment (bitrise-tom) [inline roles/bitrise-cli/tests/test_bitrise_cli.py:1]: these tests should check the version output from the tools, not just the existence of the binary

File: roles/go/tests/test_go.py [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+def test_go_bin_exists(host):
+    bitrise = host.file("/home/linuxbrew/go/bin")
+    assert bitrise.is_directory
+    assert bitrise.user == "linuxbrew"
+
+def test_go_pkg_exists(host):
+    bitrise = host.file("/home/linuxbrew/go/pkg")
+    assert bitrise.is_directory
+    assert bitrise.user == "linuxbrew"
```
Comment (bitrise-tom) [inline roles/go/tests/test_go.py:19]: add a newline

File: roles/go/tests/test_go.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+def test_if_go_exists(host):
+    assert host.exists("go")
+
+def test_go_src_exists(host):
+    bitrise = host.file("/home/linuxbrew/go/src")
+    assert bitrise.is_directory
+    assert bitrise.exists
+    assert bitrise.user == "linuxbrew"
+
+def test_go_bin_exists(host):
```
Comment (bitrise-tom) [inline roles/go/tests/test_go.py:11]: should we check we have the correct go binary installed?

File: roles/remote-access/tests/test_remote_access.py [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    assert ngrok.group == "wheel"
+    assert ngrok.mode == 0o755
+
+def test_bootstrap_script_exists(host):
+    bootstrap_script = host.file("/opt/bitrise/bootstrap_remote_access.sh.tmpl")
+    assert bootstrap_script.user == "linuxbrew"
+    assert bootstrap_script.mode == 0o777
+
+# Test ngrok
+def test_ngrok_exists(host):
```
Comment (bitrise-tom) [inline roles/remote-access/tests/test_remote_access.py:20]: test the version? It's actually nice to be warned if it changes

**Review by bitrise-tom — CHANGES_REQUESTED** [review]

File: roles/go/tests/test_go.py (hunk 8 above)
Comment (fehersanyi-bitrise) [reply roles/go/tests/test_go.py:11]: how do you imagine that?🧐

File: roles/bitrise-cli/tasks/main.yml (hunk 5 above)
Comment (bitrise-tom) [reply roles/bitrise-cli/tasks/main.yml:6]: why is this resolved?

File: roles/bitrise-cli/tasks/main.yml (hunk 4 above)
Comment (bitrise-tom) [reply roles/bitrise-cli/tasks/main.yml:21]: why is this resolved?

File: roles/bitrise-cli/tests/test_bitrise_cli.py (hunk 6 above)
Comment (bitrise-tom) [reply roles/bitrise-cli/tests/test_bitrise_cli.py:1]: @fehersanyi-bitrise why are you resolving these comments without either fixing or replying?

File: roles/remote-access/tests/test_remote_access.py (hunk 9 above)
Comment (fehersanyi-bitrise) [reply roles/remote-access/tests/test_remote_access.py:20]: we install the latest stable, I'm not sure if a failing test would be the best way to check if it's updated

File: roles/bitrise-cli/tasks/main.yml (hunk 5 above)
Comment (fehersanyi-bitrise) [reply roles/bitrise-cli/tasks/main.yml]: it was removed, I just forget to push it yesterday night

File: roles/go/tests/test_go.py (hunk 8 above)
Comment (bitrise-tom) [reply roles/go/tests/test_go.py:12]: ok - i see we just brew install the latest. This test will be enough for now

File: roles/remote-access/tests/test_remote_access.py (hunk 9 above)
Comment (bitrise-tom) [reply roles/remote-access/tests/test_remote_access.py:20]: `/bin.equinox.io/c/gDfFGFRN2Jh/ngrok-link-stable-darwin-amd64.zip`
ngrok comes from a hash URL, so version checking won't be necessary, but can do a similar check to the bitrise clie

```ngrok -v
ngrok version 2.3.34
```

**Review by bitrise-tom — APPROVED** [review]

**Review by Kanan-Bitrise — APPROVED** [review]

**Review by peterheja-bitrise — APPROVED** [review]
lgtm

**Review by MrBlaise — APPROVED** [review]
lgtm

**Pull request merged**